#! /bin/sh /usr/share/dpatch/dpatch-run
## 52_use_dmix_dsnoop_for_maestro3.dpatch by Daniel T Chen <crimsun@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use dmix/dsnoop for maestro3
## DP: http://hg-mirror.alsa-project.org/alsa-lib?cs=c64338235908;style=raw

@DPATCH@
diff -urNad alsa-lib-1.0.13~/src/conf/cards/Maestro3.conf alsa-lib-1.0.13/src/conf/cards/Maestro3.conf
--- alsa-lib-1.0.13~/src/conf/cards/Maestro3.conf	2006-09-29 07:42:56.000000000 -0400
+++ alsa-lib-1.0.13/src/conf/cards/Maestro3.conf	2007-02-04 18:38:13.000000000 -0500
@@ -11,3 +11,27 @@
 	card $CARD
 	device 0
 }	
+
+# default with dmix/dsnoop
+Maestro3.pcm.default {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type asym
+	playback.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dmix:" $CARD ]
+		}
+	}
+	capture.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dsnoop:" $CARD ]
+		}
+	}
+}
+
